### PR TITLE
Add Style/IsAFilter rule

### LINKED ISSUE #175: [Feature Request] .reject(Nil)
I recently used the code
```cr
...
  .reject(&.nil?)
  .map(&.not_nil!)
```
but I learned (ty ary) there is a much better way.
```cr
...
  .reject(Nil)
```

### PR DESCRIPTION
**Pull request by Sija**
Resolves #175

### CODE REVIEW

File: src/ameba/rule/style/is_a_filter.cr [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      return unless (block = node.block)
+      return unless (body = block.body).is_a?(Crystal::IsA)
+      return unless (path = body.const).is_a?(Crystal::Path)
+      return unless body.obj.is_a?(Crystal::Var)
+
+      name = path.names.join("::")
+      name = "::#{name}" if path.global? && !body.nil_check?
+
+      end_location = node.end_location
+      if !end_location || end_location.try(&.column_number.zero?)
```
Comment (veelenga) [inline src/ameba/rule/style/is_a_filter.cr:60]: haven't seen the column number to be zero. Can it be so?

Comment (Sija) [reply src/ameba/rule/style/is_a_filter.cr:60]: I'm not sure as to how but I've seen it happen.

Comment (Sija) [reply src/ameba/rule/style/is_a_filter.cr:60]: IIRC it occurred when expression had no parenthesis.

Comment (veelenga) [reply src/ameba/rule/style/is_a_filter.cr:60]: might be a bug in the compiler. But anyway, better to have a condition here for safety.

**Review by veelenga — APPROVED** [review]

**Pull request merged**